### PR TITLE
[DEV-3063] bankAccount bankData sync

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
- [ ] add migration

File: src/subdomains/generic/user/models/bank-data/bank-data.entity.ts [hunk 1]
```diff
@@ -20,6 +21,10 @@ export enum BankDataVerificationError {
 }
 
 @Entity()
+@Index((bankData: BankData) => [bankData.iban, bankData.userData], {
+  unique: true,
+  where: `type = '${BankDataType.USER}'`,
+})
```
Comment (davidleomay) [inline src/subdomains/generic/user/models/bank-data/bank-data.entity.ts:27]: Does this interfere with current data?

File: src/subdomains/supporting/bank/bank-account/bank-account.entity.ts [hunk 2]
```diff
@@ -36,6 +36,9 @@ export class BankAccount extends IEntity implements BankAccountInfos {
   @Column({ length: 256 })
   iban: string;
 
+  @Column({ nullable: true })
+  synced: boolean;
```
Comment (davidleomay) [inline src/subdomains/supporting/bank/bank-account/bank-account.entity.ts:40]: `default: false`?

File: src/subdomains/generic/user/models/bank-data/bank-data.entity.ts (hunk 1 above)
Comment (Yannick1712) [reply src/subdomains/generic/user/models/bank-data/bank-data.entity.ts:27]: Jap that should not be a problem 👍  I checked it manually in prd DB. Looks good 👍

File: src/subdomains/supporting/bank/bank-account/bank-account.entity.ts (hunk 2 above)
Comment (Yannick1712) [reply src/subdomains/supporting/bank/bank-account/bank-account.entity.ts:40]: I think, we don't need it. The column is only temporary for the sync 👍

**Review by davidleomay — APPROVED** [review]
- [x] Add migration

**Pull request merged**